frontend/www/js/omegaup/components/contest/AddProblem.test.ts

```diff
@@ -61,8 +76,165 @@ describe('AddProblem.vue', () => {
       },
     });
 
-    expect(wrapper.find('td[data-remove-problem] button').text()).toContain(
-      '🚫',
+    expect(
+      wrapper.find('button[data-remove-problem-disabled="problem"]'),
+    ).toBeTruthy();
+  });
+
+  it('Should update a problem in the list', async () => {
+    const wrapper = shallowMount(contest_AddProblem, {
+      propsData: {
+        contestAlias: 'testContestAlias',
+        initialPoints: 100,
+        initialProblems: [
+          { ...problem, has_submissions: false },
+          { ...problem, alias: 'problem_2', letter: 'B', title: 'Problem 2' },
+        ],
+      },
+    });
+
+    expect(wrapper.find('form').text()).not.toContain(
+      T.contestAddproblemChooseVersion,
+    );
+    expect(wrapper.find('form').text()).not.toContain(T.wordsPoints);
+    expect(wrapper.find('form').text()).not.toContain(
+      T.contestAddproblemProblemOrder,
+    );
+
+    expect(wrapper.find('button[data-update-problem="problem"]')).toBeTruthy();
+    await wrapper
+      .find('button[data-update-problem="problem"]')
+      .trigger('click');
+
+    expect(wrapper.find('form').text()).toContain(
+      T.contestAddproblemChooseVersion,
     );
+    expect(wrapper.find('form').text()).toContain(T.wordsPoints);
+    expect(wrapper.find('form').text()).toContain(
+      T.contestAddproblemProblemOrder,
+    );
+    expect(wrapper.find('button.add-problem').text()).toBe(
+      T.wordsUpdateProblem,
+    );
+
+    const updatedProblem = {
+      order: '2',
+      points: '98',
+    };
+    wrapper.setData(updatedProblem);
+
+    await wrapper.find('button.add-problem').trigger('click');
+    expect(wrapper.emitted('add-problem')).toBeDefined();
+    expect(wrapper.emitted('add-problem')).toEqual([
+      [
+        {
+          isUpdate: true,
+          problem: {
+            ...updatedProblem,
+            alias: 'problem',
+            commit,
+          },
+        },
+      ],
+    ]);
+  });
+
+  it('Should update non-latest version for a problem in the list', async () => {
+    const alternativeCommit = 'b6918b04024976471c4cac1d2efeb81021d93345';
+    const versionLog: types.ProblemVersion[] = [
+      { ...revision },
+      { ...revision, commit: alternativeCommit },
+    ];
+
+    const wrapper = mount(contest_AddProblem, {
+      propsData: {
+        contestAlias: 'testContestAlias',
+        initialPoints: 100,
+        initialProblems: [
+          { ...problem, has_submissions: false },
+          { ...problem, alias: 'problem_2', letter: 'B', title: 'Problem 2' },
+        ],
+      },
+    });
+
+    await wrapper
+      .find('button[data-update-problem="problem"]')
+      .trigger('click');
+
+    wrapper.setData({ selectedRevision: revision, versionLog });
```

Comment:
```suggestion
    wrapper.setData({ versionLog });
```

hay que ver cómo hacer que el `selectedRevision` se actualice dentro del componente en vez de pasárselo desde afuera.

Reply:
¿Crees que valga la pena obtener las versiones de todos los problemas desde el payload, y conforme se vayan agregando nuevos problemas, estos se almacenen en un store de vuex?

Reply:
Ah, y ya vi que no es necesario el store porque cada que se agrega/actualiza un problema, mandamos llamar `api.Contest.Problems`, así que también ahí podría agregar las versiones por problema.

Reply:
Listo, con los cambios ya se actualiza desde el mismo componente